json/src/bytes.rs

```diff
@@ -33,38 +32,54 @@ impl Bytes {
     pub fn new(v: Vec<u8>) -> Self {
         Bytes(v)
     }
-}
 
-impl From<Bytes> for Vec<u8> {
-    fn from(b: Bytes) -> Self {
-        b.0
+    /// Convert back to vector
+    pub fn into_vec(self) -> Vec<u8> {
+        self.0
     }
-}
-
-impl Deref for Bytes {
-    type Target = [u8];
 
-    fn deref(&self) -> &[u8] {
-        &self.0
+    pub fn without_prefix(&self) -> WithoutPrefix<&Bytes> {
+        WithoutPrefix(self)
     }
 }
 
 impl FromStr for Bytes {
-    type Err = Never;
+    type Err = BytesError;
```

Comment:
Why not using `FromHexError` directly?

Reply:
I thought to reuse dependee package's error is somewhat leaky, but it seems to be okay.

Reply:
It's okay to wrap the type defined in the other package if new type derives the same traits of the inner type.